### ISSUE TITLE
Linking PySP stage variables with variables defined in the block

### ISSUE THREAD

**Issue by wanqiuchansheng** [Pyomo/pyomo#714]
The following is the code. but it doesnt work!
```
from pyomo.environ import *

scenario_data = {}
scenario_data['Scenario1'] = (2.1, 23.5)
scenario_data['Scenario2'] = (0.2, 4.2)
scenario_data['Scenario3'] = (0.6, 0.1)
scenario_data['Scenario4'] = (1.1, -4.4)


def pysp_scenario_tree_model_callback():
    from pyomo.pysp.scenariotree.tree_structure_model \
        import CreateConcreteTwoStageScenarioTreeModel

    st_model = CreateConcreteTwoStageScenarioTreeModel(len(scenario_data))

    first_stage = st_model.Stages.first()
    second_stage = st_model.Stages.last()

    # First Stage
    st_model.StageCost[first_stage] = 'FirstStageCost'
    st_model.StageVariables[first_stage].add('x')
    st_model.StageVariables[first_stage].add('v1[*]')
    # not work either!
    # st_model.StageDerivedVariables[first_stage].add('v1.xd')
    # st_model.StageDerivedVariables[first_stage].add('v1.xde')

    # Second Stage
    st_model.StageCost[second_stage] = 'SecondStageCost'
    st_model.StageVariables[second_stage].add('y')
    st_model.StageVariables[second_stage].add('v2[*]')
    # not work either!
    # st_model.StageDerivedVariables[second_stage].add('v2.yd')
    # st_model.StageDerivedVariables[second_stage].add('v2.yde')

    return st_model


# Creates an instance for each scenario
def pysp_instance_creation_callback(scenario_name, node_names):
    a, b = scenario_data[scenario_name]

    model = ConcreteModel()
    model.x = Var()
    model.y = Var()
    model.FirstStageCost = Expression(expr=model.x ** 2)
    model.SecondStageCost = Expression(expr=a * (model.y - b) ** 2)
    model.obj = Objective(expr=model.FirstStageCost + \
                               model.SecondStageCost)

    def _v1(b):
        b.xd = Var()
        b.xde = Expression(expr=model.x)

    model.v1 = Block(rule=_v1)
    v1 = model.v1

    model.v2 = Block()
    v2 = model.v2
    model.v2.yd = Var()
    model.v2.yde = Expression(expr=model.y - b)

    model.c = ConstraintList()
    model.c.add(model.y == model.x)
    model.c.add(v1.xd == model.x)
    model.c.add(v2.yd == model.y - b)

    return model
```
the exception is below.
```
Traceback (most recent call last):
  File "/home/administrator/project/python/gym-rl/pyomotools/test/pysp/simple_quadratic/run.py", line 12, in <module>
    stsolver = rapper.StochSolver("ReferenceModel2.py")
  File "/home/administrator/project/python/gym-rl/pyomotools/rapper.py", line 113, in __init__
    self.scenario_tree.linkInInstances(instances)
  File "/home/administrator/.conda/envs/python36/lib/python3.6/site-packages/pyomo/pysp/scenariotree/tree_structure.py", line 1767, in linkInInstances
    initialize_solution_data=initialize_solution_data)
  File "/home/administrator/.conda/envs/python36/lib/python3.6/site-packages/pyomo/pysp/scenariotree/tree_structure.py", line 1881, in populateVariableIndicesAndValues
    initialize_solution_data=initialize_solution_data)
  File "/home/administrator/.conda/envs/python36/lib/python3.6/site-packages/pyomo/pysp/scenariotree/tree_structure.py", line 492, in populateVariableIndicesAndValues
    name_index_to_id_map=name_index_to_id_map)
  File "/home/administrator/.conda/envs/python36/lib/python3.6/site-packages/pyomo/pysp/scenariotree/tree_structure.py", line 284, in updateVariableIndicesAndValues
    scenario_instance.name))
RuntimeError: The component=v1 associated with stage=Stage1 is not present in instance=Scenario1
```

**Comment by blnicho** [comment]
@wanqiuchansheng I'm unable to reproduce this error using your model code. What versions of Python and Pyomo are you using?

**Comment by wanqiuchansheng** [comment]
> @wanqiuchansheng I'm unable to reproduce this error using your model code. What versions of Python and Pyomo are you using?

python3.6 and pyomo 5.5.1，i  install  it from github

**Comment by wanqiuchansheng** [comment]
hi @blnicho, thanks for your reply, i found the reason. It may be due to rapper's bug.

https://github.com/Pyomo/pyomo/blob/f4a81a2ce4c2d928c85b4fdfea32f8e081a4d617/pyomo/pysp/util/rapper.py#L72-L109

i create a new model file with name "ReferenceModel2.py"， and want to use default treemodel defined in ReferenceModel2.py return by pysp_scenario_tree_model_callback function.  However, rapper refer to the ReferenceModel.py when tree_model is None whatever fsfile is setting.

**Comment by blnicho** [comment]
@wanqiuchansheng how are you calling PySP to solve your model?

**Comment by blnicho** [comment]
@wanqiuchansheng I think this was resolved in #727 so I'm closing this issue. Please re-open if you're still having trouble.
